evap/evaluation/models.py

```diff
@@ -2187,11 +2250,31 @@ def send_participant_publish_notifications(cls, evaluations, template=None):
             template.send_to_user(participant, {}, body_params, use_cc=True)
 
     @classmethod
-    def send_textanswer_reminder_to_user(cls, user: UserProfile, evaluation_url_tuples: list[tuple[Evaluation, str]]):
+    def send_textanswer_reminder_to_user(
+        cls, user: UserProfile, evaluation_url_tuples: list[tuple[Evaluation, str]]
+    ) -> None:
         body_params = {"user": user, "evaluation_url_tuples": evaluation_url_tuples}
         template = cls.objects.get(name=cls.TEXT_ANSWER_REVIEW_REMINDER)
         template.send_to_user(user, {}, body_params, use_cc=False)
 
+    @classmethod
+    def send_grade_reminder(
+        cls,
+        recipient_email: str,
+        semester: Semester,
+        responsibles_and_courses_without_final_grades: Iterable[tuple[UserProfile, list[Course]]],
```

Comment:
Can things break if this `Iterable` is a generator that is exhausted after the first time iterating through all items? I suppose a template could iterate once for each language. Maybe we should be more conservative and go with `list`? (Or is my interpretation of `Iterable` incorrect?)

Reply:
We agreed on: `Iterable` indicates that it suffices that something can be iterated once. We will use `Collection` when something needs to be iterable multiple times.